app/models/identity.rb

```diff
@@ -2,6 +2,10 @@ class Identity < ActiveRecord::Base
   include NonNullUuid
 
   belongs_to :user
+  belongs_to :sp,
```

Comment:
I can refactor our app to use the association if you want :)

I disagree. I think of the `Identities` table as a join between the `User` and the `ServiceProvider`, we expose a user to a service provider and create an identity.